transaction/core/src/onetime_keys.rs

```diff
@@ -1,124 +1,174 @@
+//!
+//! ## User account keys (a,b)
+//! To begin, a user creates unique account keys `(a,b)`, where `a` is the private view key and
+//! `b` is the private spend key. The corresponding public keys are `A = a*G` and `B = b*G`, where
+//! `G` is the Ristretto base point. The keys `a`, `b`, `A`, and `B` "stay in the user's wallet":
+//! they are not shared with other users and they do not appear in the ledger.
+//!
+//! ## Creating the i^th subaddress (C_i, D_i)
+//! Instead, when a user wishes to receive MobileCoin, they compute a pair of public keys
+//!
+//!    `D_i = B + Hs( a | i ) * G`
+//!    `C_i = a * D`
+//!
+//! where `Hs` denotes an appropriately domain-separated hash function that returns a scalar.
+//! `C_i` is called the subaddress public view key; `D_i` is the public subadress spend key.
+//! The `subaddress index` `i` allows the user to generate many distinct subaddresses from a single
+//! address.
+//!
+//! See the `account_keys` crate for more about account keys and subaddresses.
+//!
+//! ## Sending MobileCoin to a subaddress (C,D)
+//! To send MobileCoin to a recipient's subaddress (C,D), the sender generates a unique random
+//! number `r`, and creates the following public keys and includes them in a transaction output:
+//!
+//!    `onetime_public_key = Hs( r * C ) * G + D`
+//!    `tx_public_key = r * D`
+//!
+//! The `onetime_public_key` is sometimes called `target_key`.
+//!
+//! ## Identifying an output sent to your subaddress (C_i, D_i).
+//! If you are the recipient of an output, even though you don’t know the random number `r`
+//! used in the output's tx_pub_key, you can use the fact that `a * rD_i = r * aD_i = rC_i` and
+//! compute the value
+//!
+//!    `Hs( a * tx_public_key ) * G + D_i`.
+//!
+//! If this value equals the output's onetime_key, then the output was sent to your i^th subaddress.
+//!
+//! ## Spending MobileCoin sent to your subaddress (C_i, D_i)
+//! To spend an output sent to your i^th subaddress, compute the onetime private key:
+//!
+//!     `onetime_private_key = Hs(a * tx_public_key) + d`
+//!                         `= Hs(a * tx_public_key) + b + Hs( a | i )`
 //!
 //! # References
 //! * [CryptoNote Whitepaper, Sections 4.3 and 4.4](https://cryptonote.org/whitepaper.pdf)
 
 #![allow(non_snake_case)]
 
-use crate::domain_separators::{HASH_TO_POINT_DOMAIN_TAG, HASH_TO_SCALAR_DOMAIN_TAG};
+use crate::domain_separators::HASH_TO_SCALAR_DOMAIN_TAG;
 use blake2::{Blake2b, Digest};
 use curve25519_dalek::{
     constants::RISTRETTO_BASEPOINT_POINT, ristretto::RistrettoPoint, scalar::Scalar,
 };
 use mc_account_keys::{PublicAddress, ViewKey};
 use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
-use mc_util_from_random::FromRandom;
-use rand_core::{CryptoRng, RngCore};
 
 const G: RistrettoPoint = RISTRETTO_BASEPOINT_POINT;
 
-/// Applies a hash function and returns a RistrettoPoint.
-pub fn hash_to_point(ristretto_public: &RistrettoPublic) -> RistrettoPoint {
```

Comment:
This moved to the ring_signature module because it is only used when working with key images.